File: week_1/day-2-prefix-sum-hashing/prefix/Why_is_prefix_sum_useful.py
arr = [2, 4, 6, 8]

prefix = [0] * len(arr)
prefix[0] = arr[0]

for i in range(1, len(arr)):
    prefix[i] = prefix[i - 1] + arr[i]

# sum from index 1 to 3
l, r = 1, 3
range_sum = prefix[r] - prefix[l - 1]

File: week_1/day-2-prefix-sum-hashing/prefix/Why_is_prefix_sum_useful.js
const arr = [2, 4, 6, 8];

// Build prefix sum array
const prefix = new Array(arr.length);
prefix[0] = arr[0];

for (let i = 1; i < arr.length; i++) {
    prefix[i] = prefix[i - 1] + arr[i];
}

// Sum from index 1 to 3
const l = 1, r = 3;
const rangeSum = prefix[r] - prefix[l - 1];

console.log(rangeSum); // Output: 18
